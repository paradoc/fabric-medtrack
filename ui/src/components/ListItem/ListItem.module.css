.listItem {
  box-shadow:
    0 3px 6px rgba(0,0,0,0.15),
    0 2px 4px rgba(0,0,0,0.12);
  height: 64px;
  display: grid;
  place-items: center;
  grid-template-columns: auto 1fr;
  column-gap: 12px;
  padding: 8px 24px;
  border-radius: 12px;
}

.listItem > input[type="checkbox"] {
  accent-color: aquamarine;
  width: 20px;
  height: 20px;
}

.listItem > label {
  display: grid;
  row-gap: 8px;
  width: 100%;
}

.listItem > label > strong {
  font-weight: bold;
  font-size: 20px;
  color: #333;
}

.listItem > label > span {
  font-size: 14px;
  color: #999;
}