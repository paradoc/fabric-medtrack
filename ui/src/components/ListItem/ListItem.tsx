import React, { useEffect, useRef } from 'react'
import { Medication } from '../../views/Collector'

import styles from './ListItem.module.css'

interface Props {
  data: Medication
  onClick: (a: boolean) => void
  shouldReset: boolean
}

export default function ListItem({
  data: { brand_name, generic_name },
  onClick,
  shouldReset,
}: Props) {
  const id = `${generic_name}-${brand_name}`
  const ref = useRef<HTMLInputElement>(null)

  useEffect(() => {
    if (shouldReset && ref.current !== null) {
      ref.current.checked = false
    }
  }, [ref, shouldReset])

  return (
    <div className={styles.listItem}>
      <input
        type="checkbox"
        id={id}
        onChange={(e) => onClick(e.currentTarget.checked)}
        ref={ref}
        data-testid="checkbox"
      />
      <label htmlFor={id}>
        <strong>{generic_name}</strong> <span>{brand_name}</span>
      </label>
    </div>
  )
}
