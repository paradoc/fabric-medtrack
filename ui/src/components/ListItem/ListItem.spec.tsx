import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import produce from 'immer'
import ListItem from './ListItem'

const mockClick = jest.fn()

describe('ListItem', () => {
  test('should match the snapshot and confirm the click button behavior', () => {
    const props = {
      data: {
        brand_name: 'test-brand',
        generic_name: 'test-generic',
        frequency: '',
        end_after_n: 1,
      },
      onClick: mockClick,
      shouldReset: true,
    }

    const { rerender, container, getByTestId } = render(<ListItem {...props} />)
    rerender(
      <ListItem
        {...produce(props, (data) => {
          data.shouldReset = false
        })}
      />
    )
    fireEvent.click(getByTestId('checkbox'))
    expect(mockClick).toHaveBeenCalled()
    expect(container).toMatchSnapshot()
  })
})
