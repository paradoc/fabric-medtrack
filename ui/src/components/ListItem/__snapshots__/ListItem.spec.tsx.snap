// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match the snapshot and confirm the click button behavior 1`] = `
<div>
  <div
    class="listItem"
  >
    <input
      data-testid="checkbox"
      id="test-generic-test-brand"
      type="checkbox"
    />
    <label
      for="test-generic-test-brand"
    >
      <strong>
        test-generic
      </strong>
       
      <span>
        test-brand
      </span>
    </label>
  </div>
</div>
`;
